Core.ApplicationServices/LocalOptions/GenericLocalOptionsService.cs

```diff
@@ -0,0 +1,227 @@
+                            _localOptionRepository.Save();
+                            return GetByOrganizationUuidAndOptionUuid(organizationUuid, globalOptionUuid);
+                        })
+                );
+        }
+
+        private Result<TLocalOptionType, OperationError> ValidateModify(TLocalOptionType localOption)
+        {
+            return _authorizationContext.AllowModify(localOption)
+                ? localOption
+                : new OperationError($"User not allowed to modify local option with id: {localOption.Id}", OperationFailure.Forbidden);
+        }
+
+        public Result<TOptionType, OperationError> DeleteLocalOption(Guid organizationUuid, Guid globalOptionUuid)
+        {
+            return GetLocalOptionByOrganizationUuidAndGlobalOptionUuid(organizationUuid, globalOptionUuid)
+                .Match(localOption => _authorizationContext.AllowDelete(localOption)
+                    ? Result<TLocalOptionType, OperationError>.Success(localOption)
+                    : new OperationError($"User not allowed to delete local option with global option uuid: {globalOptionUuid}",
+                        OperationFailure.Forbidden),
+                    () => new OperationError($"Local option in organization with uuid: {organizationUuid} with global option uuid: {globalOptionUuid} was not found", OperationFailure.NotFound))
+                .Bind(localOption =>
+                {
+                    localOption.Deactivate();
+                    _localOptionRepository.Update(localOption);
+                    _domainEvents.Raise(new EntityUpdatedEvent<TLocalOptionType>(localOption));
+
+                    _localOptionRepository.Save();
+                    return GetByOrganizationUuidAndOptionUuid(organizationUuid, globalOptionUuid);
+                });
+        }
+
+        private Result<int, OperationError> ResolveOrganizationId(Guid organizationUuid)
+        {
+            return _identityResolver.ResolveDbId<Organization>(organizationUuid)
+                .Match<Result<int, OperationError>>(id => id,
+                    () => new OperationError($"Organization with uuid: {organizationUuid} not found",
+                        OperationFailure.NotFound));
+        }
+
+        private Result<int, OperationError> ResolveOrganizationIdAndValidateCreate(Guid organizationUuid)
+        {
+            return ResolveOrganizationId(organizationUuid)
+                .Bind(id => _authorizationContext.AllowCreate<TLocalOptionType>(id)
+                    ? Result<int, OperationError>.Success(id)
+                    : new OperationError(
+                        $"User not allowed to create Local options for organization with uuid: {organizationUuid}",
+                        OperationFailure.Forbidden));
+        }
+
+
+
+        private Maybe<TLocalOptionType> GetLocalOptionByOrganizationUuidAndGlobalOptionUuid(Guid organizationUuid, Guid globalOptionUuid)
```

Comment:
I think this method name can be made more conscise. I think 'GetLocalOption' is fine since the type signature already communicates that it's using organizationUuid and globalOptionUuid to find it.